src/compas_rhino/geometry/brep/brep.py

```diff
@@ -183,6 +220,88 @@ def volume(self):
         if self._brep:
             return self._brep.GetVolume()
 
+    @property
+    def centroid(self):
+        assert self._brep
+        centroid = Rhino.Geometry.AreaMassProperties.Compute(self._brep).Centroid
+        return Point(*centroid)
+
+    @property
+    def curves(self):
+        assert self._brep
+        return [RhinoNurbsCurve.from_native(c.ToNurbsCurve()) for c in self._brep.Curves3D]
+
+    @property
+    def is_closed(self):
+        assert self._brep
+        return self._brep.IsSolid
+
+    @property
+    def is_compound(self):
+        # TODO: clarify. according to the internets compound brep is actually a container for several breps, not sure that's possible with a Rhino Brep.
+        return False
+
+    @property
+    def is_compoundsolid(self):
+        # TODO: see above
+        return False
+
+    @property
+    def is_convex(self):
+        raise NotImplementedError("Convexity check is not implemented for Rhino Breps.")
+
+    @property
+    def is_infinite(self):
+        # TODO: what does this exactly mean? couldn't find in the Rhino API
+        raise NotImplementedError
+
+    @property
+    def is_orientable(self):
+        assert self._brep
+        return self._brep.SolidOrientation in (Rhino.Geometry.BrepSolidOrientation.Inward, Rhino.Geometry.BrepSolidOrientation.Outward)
+
+    @property
+    def is_shell(self):
+        # not sure how to get this one
+        raise NotImplementedError
+
+    @property
+    def is_surface(self):
+        assert self._brep
+        return self._brep.IsSurface
+
+    @property
+    def is_valid(self):
+        assert self._brep
+        return self.IsValid
+
+    @property
+    def orientation(self):
+        assert self._brep
+        # TODO: align this with compas.geometry.BrepOrientation
+        return self._brep.SolidOrientation
+
+    @property
+    def shells(self):
+        # TODO: can create shell from brep but have to specify which faces to eliminate in order to hollow out the brep, doesn't seem like the intention.
+        # TODO: is this about traversing a compound brep?
+        raise NotImplementedError("Shells are not implemented for Rhino Breps.")
+
+    @property
+    def solids(self):
```

Comment:
also OCC terminology. every solid is bound by a shell. not every shell is a solid...